cwms-data-api/src/main/java/cwms/cda/data/dao/location/kind/LockDao.java

```diff
@@ -0,0 +1,321 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package cwms.cda.data.dao.location.kind;
+
+import static cwms.cda.data.dao.location.kind.LocationUtil.getLocationRef;
+import static cwms.cda.data.dao.location.kind.LocationUtil.getLookupType;
+import static java.util.stream.Collectors.toList;
+import static org.jooq.impl.DSL.asterisk;
+import static org.jooq.impl.DSL.field;
+import static org.jooq.impl.DSL.name;
+import static org.jooq.impl.DSL.table;
+import static org.jooq.impl.DSL.val;
+
+import cwms.cda.api.enums.UnitSystem;
+import cwms.cda.api.errors.NotFoundException;
+import cwms.cda.data.dao.DeleteRule;
+import cwms.cda.data.dao.JooqDao;
+import cwms.cda.data.dao.LocationsDao;
+import cwms.cda.data.dao.LocationsDaoImpl;
+import cwms.cda.data.dto.CwmsId;
+import cwms.cda.data.dto.Location;
+import cwms.cda.data.dto.LookupType;
+import cwms.cda.data.dto.location.kind.Lock;
+import cwms.cda.data.dto.location.kind.LockLocationLevelRef;
+import java.time.ZoneId;
+import java.util.List;
+import org.jooq.DSLContext;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.jooq.SelectConditionStep;
+import org.jooq.Table;
+import org.jooq.impl.DSL;
+import usace.cwms.db.jooq.codegen.packages.CWMS_LOCK_PACKAGE;
+import usace.cwms.db.jooq.codegen.tables.AV_LOCK;
+import usace.cwms.db.jooq.codegen.udt.records.LOCK_OBJ_T;
+
+public final class LockDao extends JooqDao<Lock> {
+    private static final AV_LOCK view = AV_LOCK.AV_LOCK;
+
+    public LockDao(DSLContext dsl) {
+        super(dsl);
+    }
+
+    public List<Lock> retrieveLockCatalog(CwmsId projectId) {
+        return connectionResult(dsl, c -> {
+            setOffice(c, projectId.getOfficeId());
+            Result<Record> catalogResults = CWMS_LOCK_PACKAGE.call_CAT_LOCK(dsl.configuration(),
+                    projectId.getName(), projectId.getOfficeId());
+            return catalogResults.stream().map(LockDao::catMap).collect(toList());
+        });
+    }
+
+    public Lock retrieveLock(CwmsId lockId, String projectId, UnitSystem units) {
```

Comment:
Remove projectId (per discussion). 

I'm also wondering if we need to do unit conversion at all here. Doesn't look like the embankments or turbines pass in units or a unit system.

Reply:
Seems like an oversight in embankments and turbines

Reply:
We will just do the unit conversion in CDA for now rather than in the pl/sql.

Reply:
Added unit conversion to DAO